internal/view/render.go

```diff
@@ -0,0 +1,97 @@
+package view
+
+import (
+	"io"
+
+	"golang.org/x/exp/slices"
+)
+
+type Presenter struct {
+	org  string
+	path string
+
+	untested *untestedComponents
+	issues   *issues
+	summary  *summary
+}
+
+// GeneratePresenter will combine and _mutate_ data to construct views used to
+// render a report and present the issues in a sorted order, based of follow
+// precedence: severity, vulnerability reference, first package in introducedBy.
+//
+// Notice: This function may alter incoming data, such as the order of elements
+// in the issues parameter.
+func GeneratePresenter(org, path string, untested []Component, issues []Issue, summary Summary) *Presenter {
+	untestedView, err := GenerateUntestedComponents(untested...)
+	if err != nil {
+		return nil
+	}
```

Comment:
i think if we move all of the computation to `p.Render()`, we don't have to swallow the errors and can still achieve the `*Presenter` return type.